testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/MultinodeProvider.java

```diff
@@ -95,56 +79,85 @@ public Standard(
     {
         this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
         this.portBinder = requireNonNull(portBinder, "portBinder is null");
-        this.imagesVersion = requireNonNull(environmentConfig, "environmentConfig is null").getImagesVersion();
-        this.jdkVersion = requireNonNull(jdkVersion, "jdkVersion is null");
         this.serverPackage = requireNonNull(serverPackage, "serverPackage is null");
+        this.jdkVersion = requireNonNull(jdkVersion, "jdkVersion is null");
+        this.imagesVersion = requireNonNull(environmentConfig, "environmentConfig is null").getImagesVersion();
         this.debug = debug;
-        checkArgument(serverPackage.getName().endsWith(".tar.gz"), "Currently only server .tar.gz package is supported");
     }
 
-    @Override
-    public void extendEnvironment(Environment.Builder builder)
-    {
-        builder.addContainers(createPrestoMaster(), createTestsContainer());
-        // default catalogs copied from /docker/presto-product-tests
-        builder.addConnector("blackhole");
-        builder.addConnector("jmx");
-        builder.addConnector("system");
-        builder.addConnector("tpch");
-    }
-
-    @SuppressWarnings("resource")
-    private DockerContainer createPrestoMaster()
+    /**
+     * Single node environment does not do it well enough and some issues are
+     * only exposed with multi node installations.
+     *
+     * By default we are creating single coordinator, single worker test clusters.
+     */
+    public EnvironmentExtender singleWorker()
     {
-        DockerContainer container =
-                createPrestoContainer(dockerFiles, serverPackage, jdkVersion, debug, "ghcr.io/trinodb/testing/centos7-oj11:" + imagesVersion, COORDINATOR)
-                        .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath("common/standard/access-control.properties")), CONTAINER_PRESTO_ACCESS_CONTROL_PROPERTIES)
-                        .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath("common/standard/config.properties")), CONTAINER_PRESTO_CONFIG_PROPERTIES);
-
-        portBinder.exposePort(container, 8080); // Trino default port
-        return container;
+        return multipleWorkers(1);
     }
 
-    @SuppressWarnings("resource")
-    private DockerContainer createTestsContainer()
+    public EnvironmentExtender multipleWorkers(int workersCount)
```

Comment:
nit: might be better to call this `workers(int count)` - `multipleWorkers(1)` seems wrong, but `workers(1)` is correct.
(Also, should you validate `workers > 0`?)

Reply:
I'll add validation whether count > 1